### PR TITLE
Feat/make one extension

### PR DESCRIPTION
**Pull request by martinnj-adsk**
The current setup of this repo is hard to maintain, having multiple lock files, and no consistent setup of typescript, linter or prettier, plus some js and some ts. This PR attempts to solve that by creating one extension that contains sample functionality inside of it. Also added a linter that we can use to make sure SDK updates doesn't break anything. An opportunity here could be to use this extension as a testing place that covers a large are of the API surface.

This also enables a way to quickly demo different functionality that the SDK can enable, and show what placement the extension authors should use. Each of the samples displays a tooltip that will send you to the code for that component. 

Haven't really done anything with the data providers, as I don't know how they should work and they in js, which makes it a little harder to port into this format. @ammarnaqvi-adsk and @samarv if you'd like to add your samples here as well, could you fix that? 

image

Next steps includes moving the code out of the `samples` directory but it got messy with all of the other stuff in this repo... I also think we should move the masterclass stuff out of here.

### CODE REVIEW

File: samples/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+If you wish to install the extension in your project, the extension ID is `36d95978-5b68-43c5-bfcc-952f6a97fc90`
+
+The extension exposes functionality in both the left panel, and the right side analysis panel.
+You can also open the left side as a floating panel to get a feel for how that works as well.
+If you find the functionality useful you can go directly to the source code by hovering the panel
+
+The functionality is seperated by what we think is the fitting placement for different types of functionality.
+
+- If you extension is an analysis of any sort, you should place it in the right menu
+- The left menu can contain both data extension, such as imports or contextual data, or authoring capabilities.
```
Comment (hakon-matland-adsk) [inline samples/README.md]: Instead link to https://aps.autodesk.com/en/docs/forma/v1/embedded-views/introduction/ for more information about extension placement?

File: samples/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+If you want to run this extension from your own machine you will need to [create your own extension](https://aps.autodesk.com/en/docs/forma/v1/overview/getting-started/)
+
+and make sure you have enabled both right and left panel in you configuration.
+
+![Config](extension-config.png)
+
+You also need to make sure your extension is installed in your project.
+
+To start the extension, you'll need to install dependencies with
```
Comment (hakon-matland-adsk) [inline samples/README.md]:
```suggestion
To start the extension, you'll need to install dependencies with [pnpm](https://pnpm.io/installation):
```

File: samples/README.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# Forma Extension Samples
+
+This extension contains a sample of functionality that is possible when using the [Forma embedded view SDK](https://app.autodeskforma.com/forma-embedded-view-sdk/docs/index.html). Feel free to fork or clone this repo if you wish to use some
+of its functionality or need a starting point for your extension.
+
+If you wish to install the extension in your project, the extension ID is `36d95978-5b68-43c5-bfcc-952f6a97fc90`
```
Comment (hakon-matland-adsk) [inline samples/README.md]:
```suggestion
If you wish to install the extension in your project, the unpublished extension ID `36d95978-5b68-43c5-bfcc-952f6a97fc90` is configured to be used with this sample extension running locally.
```

File: samples/src/style.css [hunk 4]
```diff
@@ -0,0 +1,51 @@
+html {
```
Comment (hakon-matland-adsk) [inline samples/src/style.css]: I notice scrollbars in right side whenever I open a sub thingy. Perhaps we should avoid that? Just put the comment in relevant file.
Screenshot 2024-04-09 at 12 03 42

File: README.md [hunk 5]
```diff
@@ -2,4 +2,4 @@
 
 Collection of open source sample extensions built on the forma extensions api.
 
-Docs: https://app.autodeskforma.com/extensions/preview/sdk.html
\ No newline at end of file
+Docs: https://app.autodeskforma.com/extensions/preview/sdk.html
```
Comment (hakon-matland-adsk) [inline README.md]: Dead link from "way-back" when analyze extensions had our own POC/preview SDK.

Should the samples directory just be top level instead? And remove the generators, data-providers and masterclass directories?

**Review by hakon-matland-adsk — COMMENTED** [review]
Great work @martinnj-adsk!
Added some inline comments on suggested improvements.

**Review by vilde-barth-adsk — APPROVED** [review]
This is so much better, great work @martinnj-adsk 🙌 🚢

**Pull request merged**